File: examples/src/main/java/de/maibornwolff/domainchoreograph/examples/helloworld/Application.java
package de.maibornwolff.domainchoreograph.examples.helloworld;

import de.maibornwolff.domainchoreograph.DomainAnalytics;
import de.maibornwolff.domainchoreograph.core.api.DomainEnvironment;

public class Application {

    public static void main(String[] args) {

        DomainAnalytics domainAnalytics = new DomainAnalytics();
        domainAnalytics.getServer().start();

        Name name = new Name("Bob");

        DomainEnvironment environment = DomainEnvironment.builder()
            .addLogger(domainAnalytics.getLogger())
            .build();
        MyChoreography choreography = environment.get(MyChoreography.class);

        HelloWorld helloWorld = choreography.calculate(name);

        System.out.println(helloWorld.getMessage());
    }
}

File: examples/src/main/java/de/maibornwolff/domainchoreograph/examples/helloworld/Application.kt
package de.maibornwolff.domainchoreograph.examples.helloworld

import de.maibornwolff.domainchoreograph.core.api.DomainEnvironment
import de.maibornwolff.domainchoreograph.domainanalytics.DomainAnalytics

object Application {

    @JvmStatic
    fun main(args: Array<String>) {

        // start analytics server
        val domainAnalytics = DomainAnalytics()
        domainAnalytics.server.start()

        // setup environment
        val environment = DomainEnvironment.builder()
            .addLogger(domainAnalytics.logger)
            .build()

        // input parameter
        val name = Name("Bob")

        // get generated implementation
        val choreography: HelloWorldChoreography = environment.get()
        val helloWorld = choreography.calculate(name)

        println(helloWorld.message)
    }
}
